MetaMask/metamask-extension
# Version 10.35.1

## Changelog

### Changed
- Store default gas settings by network ([#20576](https://github.com/MetaMask/metamask-extension/pull/20576), [#20632](https://github.com/MetaMask/metamask-extension/pull/20632))
- Add more diagnostic information upon failure ([#20595](https://github.com/MetaMask/metamask-extension/pull/20595))

### Fixed
- Fix bug resulting in custom network configuration being lost upon restart ([#20586](https://github.com/MetaMask/metamask-extension/pull/20586))
- Fix UI crash when balances are missing ([#20385](https://github.com/MetaMask/metamask-extension/pull/20385))
- Fix infinite rerender on network change while signature request is pending ([#20473](https://github.com/MetaMask/metamask-extension/pull/20473))
- Fix Dapp link on NFT import screen ([#19799](https://github.com/MetaMask/metamask-extension/pull/19799))
- Fix 'View on Opensea' link for main and testnet NFTs ([#19797](https://github.com/MetaMask/metamask-extension/pull/19797))
- Ensure chainId comparison in switchEthereumChain handler is case insensitive ([#20149](https://github.com/MetaMask/metamask-extension/pull/20149))
- Enforce user preferences in incoming transactions controller ([#19982](https://github.com/MetaMask/metamask-extension/pull/19982))

## Manual testing instructions

Aside from standard regression/exploratory testing, here are instructions on testing the changes in this release.

These PRs include good testing instructions:
* #20576
* #20473
* #19799
* #19797
* #20149
* #19982 (in the linked issue)
* #20385
  * For the "yarn start on v10.34.2" step, instead you can run `yarn start` on this RC to test that this works (or on v10.35.0 to test that it was still broken)

Additional details:
* For #20632, you can test the popup by running this script in the background process:
  ```
  chrome.storage.local.get(({ data, meta }) => chrome.storage.local.set({ data: { ...data, PreferencesController: { ...data.PreferencesController, advancedGasFee: { maxBaseFee: '', priorityFee: '' } } }, meta: {...meta, version: 92} }, () => { window.location.reload() }))
  ```
  That should trigger the migration in #20576 and trigger the what's new announcement from #20632

* For #20595, standard Sentry error testing steps apply (setup a personal Sentry account, set that as your DSN in `.metamaskrc`, and create a new build). The state snapshot should have far more properties than before, but still nothing personally identifiable.

For #20586, this can be tested by running this script in the background process:
  ```
  chrome.storage.local.get(({ data, meta }) => chrome.storage.local.set({ data: { ...data, NetworkController: { ...data.NetworkController, network: 'this should not exist' } }, meta: {...meta, version: 92} }, () => { window.location.reload() }))
  ```
  Previous to that change, this would result in a Sentry error and (theoretically, not confirmed) an error whenever a network state change was made (e.g. adding a custom network) that results in a failure to persist that state change. After this change, a migration should remove the extra state property, preventing any Sentry error or persistence failure.
